[Sophie] iP

sophiien opened the pull request:
# Duke

- text-based
- easy to learn 
- ~~FAST~~ SUPER FAST to use


All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
3. let it manage your tasks for you 😉

features 
- [x] managing tasks
- [ ] sorting tasks

Some basic Git commands are:
```
git status
git add
git commit
```

> "Your mind is for having ideas, not holding them" - David Allen [source](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#links).

Commit:
Add Gradle support

sophiien committed:
Level-1

sophiien committed:
Level-2

sophiien committed:
Level-3

sophiien committed:
Delete Task.class

sophiien committed:
Level-4

sophiien committed:
Level-4

sophiien committed:
Level-4

sophiien committed:
A-TextUiTesting

sophiien committed:
Level-5

sophiien committed:
Level-6

sophiien committed:
Refactored code

sophiien committed:
Level-7

sophiien committed:
added duke.txt

sophiien committed:
Level-8

sophiien committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

sophiien committed:
A-MoreOOP

sophiien committed:
made changes

sophiien committed:
A-Packages

sophiien committed:
made changes

sophiien committed:
A-JUnit

sophiien committed:
A-JavaDoc

sophiien committed:
A-CodingStandard

sophiien committed:
made changes

sophiien committed:
made changes

sophiien committed:
Level-9

Jacky142857 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+    private Storage storage;
+    private Tasklist tasks;
+    private Ui ui;
+    public static int j = 0;
```
Maybe a more descriptive name can be used instead of a letter 'j'.

Jacky142857 commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,18 @@
+    /**
+     * Prints welcome msg of chatbot
+     */
+    public void welcomeMsg() {
```
Names representing methods should be verbs. (According to coding standard)

Jacky142857 commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,18 @@
+    /**
+     * Prints ending msg of chatbot
+     */
+    public void endMsg() {
```
This name of the method should also be verb instead of a noun.

Jacky142857 commented on src/main/java/Duke/Event.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    /**
+     * Event toString()
```
In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... In this case, maybe you can phrase it like 'Returns a String representation of the Event object'

Jacky142857 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,210 @@
+     * @param time to format
+     * @return formatted time in 12h
+     */
+    public static String timeFormat(String time) {
```
You did a good job in manipulating and parsing the time!

Jacky142857 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,210 @@
+            System.out.println("OK, I've marked this task as not done yet:\n" + list.get(i - 1));
+        } else {
+            String name = "";
+            if (input.length == 1) {
```
Maybe in the future can add a new Exception class to handle this.

Jacky142857 commented on src/main/java/Duke/Tasklist.java:
```diff
@@ -0,0 +1,67 @@
+
+    /**
+     * Add task to the list
+     * @param command on how to manipulate list
```
The first word after @param is an exact copy of the parameter, a new sentence should start after this. For example '@param command Command on how to manipulate list'

Jacky142857 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,210 @@
+     * @param path of file
+     * @param text to add
+     * @throws IOException
+     * @returns void
```
maybe a typo here, an 's' after @return

Jacky142857 reviewed (commented):
Overall it looks nice and clean. However, there are some minor issues with your JavaDoc comments. You can use the Java coding standard for reference.

sophiien committed:
gradle

sophiien committed:
Level-10

sophiien committed:
Added javafx in build.gradle

sophiien committed:
Added javafx in build.gradle

sophiien committed:
Added JavaDocs

sophiien committed:
Changed relative path

sophiien committed:
Made changes to main.java

sophiien committed:
Contains dependencies in build.gradle

Does not contain dependencies for javaFx

If you do not bundle JavaFX with the JAR file, the application will not work in computers that don't have JavaFX installed.

sophiien committed:
Fixed Java version

sophiien committed:
A-Assertions

putaojuice commented on src/main/java/duke2/MainWindow.java:
```diff
@@ -0,0 +1,86 @@
+            userInput.clear();
+        }
+
+//        scrollPane.setContent(dialogContainer);
```
Hi Sophie, I am the one that reviewed your PR for the tutorial! Just sharing some of my thoughts that might help.

I feel that you should remove all these commented code because they are not used anymore and doesn't help when others are reading your code. Moreover, might confuse you as well! If you need to retrieve them in the future can always revisit the commits :)

putaojuice commented on src/main/java/duke2/Parser.java:
```diff
@@ -0,0 +1,228 @@
+     * @param task to create
+     * @return new task from string
+     */
+    public static Task parseStringToTask(String task) {
```
Your `parseStringToTask` method is really long and has a lot of nested if-else checks :<
Consider refactoring it!

putaojuice commented on src/main/java/duke2/Parser.java:
```diff
@@ -0,0 +1,228 @@
+     * @param input   full input from user
+     * @param path    of file from storage
+     */
+    public void commandToTask(String command, ArrayList<Task> list, String[] input, String path) {
```
Same goes for `commandToTask` method, it is really long and you should add some comments to explain the code, both for others and yourself. It will be a good practice to write comments to explain your code so that when next time you are revisiting this method, you can understand it easily.

putaojuice commented on src/main/java/duke2/Duke.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * Constructor for Duke
+     *
+     * @returns new Deadline object
```
I don't this Duke returns a new Deadline object?

putaojuice reviewed (commented):
I think your code looks good in general, just be careful of long methods as they are difficult to read and understand. Good job!

sophiien committed:
Merge pull request #1 from sophiien/branch-A-Assertions

A-Assertions

sophiien committed:
A-CodeQuality

sophiien committed:
Merge pull request #2 from sophiien/branch-A-CodeQuality

A-CodeQuality

sophiien committed:
BCD-Extension

sophiien committed:
Removed commented out code

sophiien committed:
Added JavaDocs to DialogBox

sophiien committed:
Merge pull request #3 from sophiien/Merging-PRs

Added JavaDocs to DialogBox

sophiien committed:
Ui.png

sophiien committed:
Merge pull request #4 from sophiien/Ui.png

Ui.png

sophiien committed:
A-UserGuide

sophiien committed:
Merge pull request #5 from sophiien/Published-UG

A-UserGuide

sophiien committed:
Refactored code and shortened functions

sophiien committed:
Update README.md

sophiien committed:
Edit README

sophiien committed:
Edit entrypoint for main app

sophiien committed:
Update UG

sophiien committed:
Update README.md

sophiien committed:
Update UG

sophiien committed:
Edit
